app/assets/src/components/views/compare/SamplesHeatmapView/SamplesHeatmapView.jsx

```diff
@@ -712,13 +783,52 @@ class SamplesHeatmapView extends React.Component {
     window.history.replaceState("", "", this.getUrlForCurrentParams());
   };
 
+  handleAddedTaxonChange = selectedTaxonIds => {
+    // selectedTaxonIds includes taxa that pass filters
+    // and the taxa manually added by the user.
+    let { taxonIds, addedTaxonIds, notifiedFilteredOutTaxa } = this.state;
+
+    // currentAddedTaxa is all the taxa manually added by the user.
+    let currentAddedTaxa = new Set([
+      ...[...selectedTaxonIds].filter(taxId => !taxonIds.includes(taxId)),
```

Comment:
Why do we need the `[... ]` here? 
I see that you repeat this pattern below...

Reply:
The [spread operator](https://exploringjs.com/impatient-js/ch_sets.html#missing-set-operations) is used in set operations in Javascript. (I just learned this, so if there's another way to do it please let me know! 😅 )

Reply:
I see... this the most compact way. You probably do not need the leading `...`. So, it would be `new Set([...selectedTaxonIds].filter(taxId => !taxonIds.includes(taxId))`.
But also, have in mind that for each includes you have a N operation. So it could be better to loop through the taxonIds array instead and remove them from a set.

Reply:
We could rewrite this using functions named merge, difference and intersection!

Reply:
Good point, rewrote to use those functions 😄